samza-core/src/main/java/org/apache/samza/system/inmemory/InMemorySystemFactory.java

```diff
@@ -31,20 +32,30 @@
  * Initial draft of in-memory {@link SystemFactory}. It is test only and not meant for production use right now.
  */
 public class InMemorySystemFactory implements SystemFactory {
-  private static final InMemoryManager MEMORY_MANAGER = new InMemoryManager();
+  private static final String TEST_ID = "test.id";
+  private static final ConcurrentHashMap<Integer, InMemoryManager> IN_MEMORY_MANAGERS = new ConcurrentHashMap<>();
```

Comment:
* Does the test.id have to be of `integer` type?
* Can you document the semantics of `test.id` here?  
* How does it work when multiple threads in a test share the same inmemory manager(Will each thread  have different in-memory-manager or all threads within the same test will have same inmemory manager)?

Reply:
- Is there any advantage to use data types other than integer?
- Sure will document semantics of `test.id` inmemory manager
- We are planning to create inmemory manager per test